chore: edit README (#5)
coder/exectrace: README.md

# exectrace [![Go Reference](https://pkg.go.dev/badge/cdr.dev/exectrace.svg)](https://pkg.go.dev/cdr.dev/exectrace)

Simple [eBPF](https://ebpf.io/)-based exec snooping on Linux packaged as a Go
library.

exectrace loads a pre-compiled [eBPF program](./bpf/handler.c) into the running
kernel to receive details about the `exec` family of syscalls.

## Requirements

> exectrace only support Go 1.16+ and Linux kernel 5.8+ (due to the use of

Author: ```suggestion
exectrace only supports Go 1.16+ and Linux kernel 5.8+ (due to the use of
```

## Usage

exectrace exposes a minimal API surface. Call `exectrace.New(nil)` and then

> you can start `tracer.Read()`ing events from the returned `tracer`.

Reviewer: ```suggestion
you can start `tracer.Read()`ing events from the returned `Tracer`.
```

My bad

It is important that you close the tracer to avoid leaking kernel resources,

> so we recommend implemeneting a simple signal handler like the one in this

Reviewer: ```suggestion
so we recommend implementing a simple signal handler like the one in this
```

}
```

> For a full usage example, see [exectrace](./cmd/exectrace/main.go), which is a

> comprehensive program that uses this library.

Reviewer: Maybe we shouldn't refer to the program as "exectrace" in the readme since it could be confusing, maybe just call it "a comprehensive example" and link to it

## Development

You will need the following:

> - Docker (run clang within a Docker container for reproducibility)

Reviewer: ```suggestion
- Docker (the Makefile runs clang within a Docker container for reproducibility)
```

- `golangci-lint`
- `prettier`
- `shellcheck`

Since the eBPF program is packaged as a Go library, you need to compile the
program and include it in the repo.

If you change the files in the `bpf` directory, run `make` and ensure that you
include the `.o` files you changed in your commit (CI will verify that you've
done this correctly).

> ## Status: in development

Reviewer: Do you have any ideas for other words we can use here? We want to convey that it's ready for use but still under active development

Author: What do you think about setting the status to "beta" and changing the first sentence to "While this is under active development, it is ready to use?"

Reviewer: yeah that sounds good to me

The library is currently under heavy development as we modify it to suit the
needs of Coder's [enterprise product](https://coder.com).

We plan on adding more features and fields that can be read from the API, as

> well as easier-to-use methods for filtering events (currently, you must
> implement filtering yourself).

Reviewer: ```suggestion
well as easier-to-use methods for filtering events (currently, you must
implement additional filtering yourself).
```
